### PR TITLE
add hsv support

### LINKED ISSUE #23: add HSV
```purescript
toHSVA :: Color.Color -> {h :: Number, s :: Number, v :: Number, a :: Number}
toHSVA color = case Color.toHSLA color of
  ({h, s, l, a}) ->
    let
      s' = s * (if l < 0.5 then l else 1.0 - l)
    in
      { h
      , s: 2.0 * s' / (l + s')
      , v: l + s'
      , a
      }

hsva :: Number → Number → Number → Number → Color.Color
hsva h s v a = Color.hsla h s' l a
  where
  tmp = (2.0 - s)*v
  s' = s * v / (if tmp < 1.0 then tmp else 2.0 - tmp)
  l = tmp / 2.0

hsv :: Number → Number → Number → Color.Color
hsv h s v = hsva h s v 1.0
```
adopted from https://gist.github.com/xpansive/1337890

### PR DESCRIPTION
**Pull request by safareli**
fix #23

### CODE REVIEW

File: src/Color.purs [hunk 1]
```diff
@@ -160,22 +163,40 @@ rgba' r g b a = rgba (round $ r * 255.0)
...
 hsla h s l a = HSLA h' s' l' a'
   where h' = if h == 360.0 then h else h `modPos` 360.0
         s' = clamp 0.0 1.0 s
         l' = clamp 0.0 1.0 l
         a' = clamp 0.0 1.0 a
 
--- | Create a `Color` from hue, saturation and lightness values. The hue is
--- | given in degrees, as a `Number` between 0.0 and 360.0. Both saturation and
--- | lightness are numbers between 0.0 and 1.0.
+-- | Create a `Color` from Hue, Saturation and Lightness values. The hue is
```
Comment (sharkdp) [inline src/Color.purs]: Same here and in the comments below.

File: src/Color.purs [hunk 2]
```diff
@@ -160,22 +163,40 @@ rgba' r g b a = rgba (round $ r * 255.0)
...
+-- | Create a `Color` from Hue, Saturation, Value and Alpha values. The
+-- | hue is given in degrees, as a `Number` between 0.0 and 360.0. Saturation,
+-- | Value and Alpha are numbers between 0.0 and 1.0.
+hsva :: Number → Number → Number → Number → Color
+hsva h s   0.0 a = HSLA h (s / (2.0 - s)) 0.0 a
+hsva h 0.0 1.0 a = HSLA h 0.0 1.0 a
+hsva h s'  v'  a = HSLA h s l a
+  where
+    tmp = (2.0 - s') * v'
+    s = (s' * v' / (if tmp < 1.0 then tmp else 2.0 - tmp))
```
Comment (sharkdp) [inline src/Color.purs]: the outer parens are unnecessary here. The same is true for the line below.

File: src/Color.purs [hunk 3]
```diff
@@ -160,22 +163,40 @@ rgba' r g b a = rgba (round $ r * 255.0)
 rgb' :: Number -> Number -> Number -> Color
 rgb' r g b = rgba' r g b 1.0
 
--- | Create a `Color` from hue, saturation, lightness and alpha values. The
+-- | Create a `Color` from Hue, Saturation, Lightness and Alpha values. The
```
Comment (sharkdp) [inline src/Color.purs]: Any particular reason you want to change that to capitalized form? If so, let's please keep it consistent (hue is lowercase in the second sentence)

File: src/Color.purs [hunk 4]
```diff
@@ -273,11 +294,22 @@ fromInt m = rgb r g b
...
 
--- | Convert a `Color` to its hue, saturation, lightness and alpha values. See
+-- | Convert a `Color` to its Hue, Saturation, Lightness and Alpha values. See
 -- | `hsla` for the ranges of each channel.
 toHSLA :: Color -> { h :: Number, s :: Number, l :: Number, a :: Number }
 toHSLA (HSLA h s l a) = { h, s, l, a }
 
+-- | Convert a `Color` to its Hue, Saturation, Value and Alpha values. See
+-- | `hsva` for the ranges of each channel.
+toHSVA :: Color -> {h :: Number, s :: Number, v :: Number, a :: Number}
```
Comment (sharkdp) [inline src/Color.purs]: Can we please keep the code style consistent here (spaces after `{` and before `}`)?

File: src/Color.purs [hunk 5]
```diff
@@ -273,11 +294,22 @@ fromInt m = rgb r g b
...
+toHSVA (HSLA h s   0.0 a) = { h , s: 2.0 * s / (1.0 + s) , v: 0.0 , a }
```
Comment (sharkdp) [inline src/Color.purs]: ... and no spaces before the `,` sign, please :smile:

File: test/Main.purs [hunk 6]
```diff
@@ -24,20 +25,20 @@ import Test.Unit.Main (runTest)
...
-     aE c1.b c2.b
-
-     where
-       c1 = toRGBA col1
-       c2 = toRGBA col2
+  else failure $ "\n    expected: " <> show' expected <>
+                 "\n    got:      " <> show' actual
+  where
+    show' c = case toHSLA c of
+      ({h, s, l, a}) -> cssStringRGBA c <> " " <> cssStringHSLA c
```
Comment (sharkdp) [inline test/Main.purs]: I believe the parens are not necessary here.

File: src/Color.purs (hunk 3 above)
Comment (safareli) [inline src/Color.purs]: Because `and Value values` looks better then `and value values`. will fix other occurrences.

**Comment by sharkdp** [comment]
Never mind about the travis fail. I'm still trying to get it running properly ;-)

**Pull request merged**